008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,1021 @@
+
+\subsection{Implementation}
+
+How agents rate treaties closely follows the theoretical steps from above. Every day, the agent stores the amount of food received in two arrays, one for the short and one for the long term. The short-term array is reset after every reshuffling period. When a new treaty is proposed, the agent performs the following steps to conclude whether to sign this new treaty or not.
+
+\begin{enumerate}
+    \item Checks if the trust in the proposer agent is high enough
+    \item Checks if the treaty is compatible with treaties that the agent signed already
+    \item Calculates the expected short- and long-term utility is according to \Cref{eq:expectedUtility} and using the memory of food received in the past. The utility function used depends on their current social motive.
+    \item Amplifies the utility if it is negative in accordance with loss-aversion.
+    \item Estimates the amount of food it can take under the treaty, i.e. the difference between the average amount of food it received in the past and the amount of food it needs to leave in order to comply with the treaty, and calculate the utility of taking that amount of food.
+    \item Computes the estimated short-term and long-term benefit of signing the treaty as $U(\text{sign}) - U(\text{don’t sign})$.
+    \item Chooses to focus on the long-term or short-term benefit according to \Cref{eq:shortlongweight}
+    \item Signs the treaty if it’s overall benefit is positive
+\end{enumerate}
+
+The amount of food that the “collectivist” and “selfish” agents would need to consume in order to maximise utility varies depending on the current health level. The peak of its total utility function thus needs to be able to vary, too. We account for this by introducing a scaling factor $a$ as
+
+\begin{equation}
+    U(x)=g(ax)^\frac{1}{r}-cax
+\end{equation}
+
+where
+
+\begin{equation}
+    a=\frac{1}{z}\left(\frac{cr}{g}\right)^\frac{r}{1-r}
+\end{equation}
+
+introducing the desired food intake $z$ as the maximum of the function without changing the general properties of the function, leading to:
+
+\begin{equation}
+U\left(x\right)=g\left(\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}x\right)^{\frac{1}{r}}-cx\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}    
+\end{equation}
+\subsection{Behaving according to treaties}
+
+Treaties are fundamentally built on the assumption that all agents behave truthfully. Introducing dishonesty to treaties would strap them of all their power, as there are two many possible falsities but no sufficient way of recognizing and no central authority to punish them. Hence, we do not consider deception: a decision shared by all agent teams. Since this means that an agent cannot act against any active treaties, the amount of food taken has to be chosen with care.
```

Comment:
```suggestion
Treaties are fundamentally built on the assumption that all agents behave truthfully. Introducing dishonesty to treaties would sap them of all their power, as there are too many possible falsities but no sufficient way of recognising them and no central authority to punish them. Hence, we do not consider deception. Since this means that an agent cannot act against any active treaties, the amount of food taken has to be chosen with care.
```